### DISCUSSION TITLE
Method addObserver must be call on the main thread

### DISCUSSION THREAD

**Discussion by zakaria-ist** [invertase/react-native-firebase#5712] [Q&A] [answered]
Hi All,
I am recently having this issue "Method addObserver must be call on the main thread" on the android platform when I am calling
 `await auth().verifyPhoneNumber(phoneNumber);`

![android_addObserver_issue](https://user-images.githubusercontent.com/17428587/132472844-bcb1345a-e754-43f2-9e4f-c3baa3405687.jpeg)

My build settings are as follows:

{
       buildToolsVersion = "30.0.2"
        minSdkVersion = 21
        compileSdkVersion = 31
        targetSdkVersion = 31
}

"react-native": "0.63.4"
"@react-native-firebase/app": "^8.4.7",
"@react-native-firebase/auth": "^9.3.0",

I think it started when we upgrade "compileSdkVersion" to 31. We cannot build using lower than 31 because of dependency.

I could not solve the issue.
Does anyone is having the issue or does anyone know what is causing the issue and how to solve?

**Comment by mikehardy** [comment, marked as answer]
Those versions of react-native-firebase are entirely too old to pay any attention to. Please update to current versions (we are on 12.7.5 here)

**Reply by zakaria-ist** [reply]
Thanks a lot.
